File: Lv.2/피보나치 수.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;
int dp[100001]={0,};
int fibonacci(int x){
    if(x==0) return 0;
    if(x==1) return 1;
    if(dp[x]!=0) return dp[x];
    return dp[x]=(fibonacci(x-1)+fibonacci(x-2))%1234567;
}

int solution(int n) {
    return fibonacci(n);
}

File: Lv.2/피보나치 수.js
var dp = Array(100001).fill(0);
function solution(n) {
    dp[0]=0;dp[1]=1;
    for(let i=2;i<=n;i++){
        dp[i] = (dp[i-1]+dp[i-2])%1234567;
    }
    return dp[n];
}